### ISSUE TITLE
Engine derefencing unecessary links when querying all triples in a graph in an `n3` document

### ISSUE THREAD

**Issue by jeswr** [comunica/comunica-feature-link-traversal#91]
#### Issue type:

- :bug: Bug 

____
#### Description:

I have done a writeup and minimal repro here: https://github.com/jeswr/link-traversal-unecessary-dereference

I understand that this is probably difficult to address with the current architecture of Comunica, however I do think resolving this unlocks some valuable use cases.

____

**Comment by rubensworks** [comment]
Indeed, in general, if a term is a blank node, we can indeed make some useful optimizations for traversal, since blank nodes are always document-scoped.

Not sure yet how we could achieve this yet though. We'll probably have to go beyond the current set of reachability semantics.
